src/queue/extras.rs

```diff
@@ -0,0 +1,13 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::time::Instant;
+
+/// The extras for the task cells pushed into a simple task queue.
+///
+/// [`Default::default`] can be used to create a [`SimpleQueueExtras`] for
+/// a task cell.
+#[derive(Debug, Default, Clone, Copy)]
+pub struct Extras {
```

Comment:
How to add new types of extras when we want to add a new queue type?

Reply:
Just add new fields. `Extras` should be small enough to be shared.

Reply:
https://github.com/tikv/yatp/pull/15/files#diff-15709e727979b90faec03201bc3303baR63
Current extras for leveled queue has quite a few fields and is not so small... And it may be confusing to mix different extras.

Reply:
Yes, I have looked into it. It add extra 24 bytes, which is the same size of `Option<Instant>`.